lib/logging.js

```diff
@@ -0,0 +1,56 @@
+
+'use strict';
+
+var path = require('path');
+var eachModule = require('each-module');
+var winston = require('winston');
+
+module.exports = function (config) {
+	// User-defined configurations take priority, but fallback to defaults if all seem invalid
+	var getArrayOfValidModulesByDirName = function (finalDir) {
+		var modules = [];
+		var modulePusher = function (moduleName, moduleExports, file) {
+			if (typeof moduleExports === 'function') {
+				modules.push(moduleExports);
+			} else {
+				console.warn('Invalid module dropped ' + file);
+			}
+		};
+
+		var locations = [config.path.root, config.path.shunterRoot]; // Config.path.root = users files
+		for (var i = 0; i < locations.length; i++) {
+			var localPath = path.join(locations[i], config.structure.logging, finalDir);
+			eachModule(localPath, modulePusher);
+			if (modules.length > 0) {
+				break;
+			}
+		}
+		return modules;
+	};
+
+	return {
+		getLogger: function () {
+			var validateTransports = function (arModules) {
```

Comment:
array and function. What types of things we are dealing with gets a bit hairy on the following lines where we are testing the types of things, so being explicit about the types of things we are expecting seemed appropriate here, albeit  inconsistent (as I expect you are noting).